pipeline/src/node_properties/transform_properties.rs

```diff
@@ -0,0 +1,114 @@
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum FlipDirection {
+    Horizontal,
+    Vertical,
+}
+
+#[derive(Default, Debug, Clone, PartialEq)]
+pub struct Crop {
+    /// Number of pixels to crop from the top.
+    pub top: usize,
+
+    /// Number of pixels to crop from the bottom.
+    pub bottom: usize,
+
+    /// Number of pixels to crop from the left.
+    pub left: usize,
+
+    /// Number of pixels to crop from the right.
+    pub right: usize,
+}
+
+impl FromStr for Crop {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.split(':').collect::<Vec<_>>()[..] {
+            [left, right, top, bottom] => {
+                match (left.parse(), right.parse(), top.parse(), bottom.parse()) {
+                    (Ok(left), Ok(right), Ok(top), Ok(bottom)) => Ok(Crop {
+                        left,
+                        right,
+                        top,
+                        bottom,
+                    }),
+                    _ => Err(format!("Failed to parse crop region string: {}", s)),
+                }
+            }
+            _ => Err(format!("Bad resolution format: {}", s)),
+        }
+    }
+}
+
+impl serde::ser::Serialize for Crop {
```

Comment:
Since `Serialize` is imported at the top of the file, we can shorten this (same for `Deserialize` below).

```suggestion
impl Serialize for Crop {
```